apis/core/lib/domain/csv-source/delete.ts

```diff
@@ -1,40 +1,54 @@
 import { NamedNode, Term } from 'rdf-js'
 import { ResourceStore } from '../../ResourceStore'
 import { resourceStore } from '../resources'
-import { deleteFile } from '../../storage/s3'
+import * as s3 from '../../storage/s3'
 import { schema } from '@tpluscode/rdf-ns-builders'
 import { cc } from '@cube-creator/core/namespace'
 import $rdf from 'rdf-ext'
+import * as TableQueries from '../queries/table'
 
 interface DeleteSourceCommand {
-  resource: NamedNode
+  resource: NamedNode | Term
   store?: ResourceStore
+  fileStorage?: s3.FileStorage
+  tableQueries?: Pick<typeof TableQueries, 'getLinkedTablesForSource'>
 }
 
 export async function deleteSource({
   resource,
   store = resourceStore(),
+  fileStorage = s3,
+  tableQueries: { getLinkedTablesForSource } = TableQueries,
 }: DeleteSourceCommand): Promise<void> {
-  await deleteSourceWithoutSave(resource, store)
+  await deleteSourceWithoutSave({ resource, store, fileStorage, tableQueries: { getLinkedTablesForSource } })
 
   // Save changes
   await store.save()
 }
 
-export async function deleteSourceWithoutSave(csvSourceTerm: Term, store: ResourceStore): Promise<void> {
-  if (csvSourceTerm.termType !== 'NamedNode') return
+export async function deleteSourceWithoutSave({
+  resource,
+  store = resourceStore(),
+  fileStorage = s3,
+  tableQueries: { getLinkedTablesForSource } = TableQueries,
+}: DeleteSourceCommand): Promise<void> {
+  if (resource.termType !== 'NamedNode') return
 
-  const csvSource = await store.get(csvSourceTerm)
+  const csvSource = await store.get(resource)
   if (!csvSource) return
 
-  // TODO: Delete Tables or prevent it
-  // Find related tables
-  // Delete them
+  // Remove links from tables
+  const tables = await getLinkedTablesForSource(csvSource.term)
+  for await (const table of tables) {
+    const tableGraph = await store.get(table)
+    tableGraph?.dataset.delete($rdf.quad(tableGraph.term, cc.csvSource, csvSource.term))
+  }
```

Comment:
I would shift this into table itself. At least if it was assigning `undefined`

```suggestion
  for await (const tableId of tables) {
    const table = await store.getResource<Table>(tableId)
    if (table) {
      table.csvSource = undefined
    }
  }
```

Which also reveals that `Table#csvSource` is not typed to allow falsy values. Removing this link without adjust the interface risks undetected runtime errors